File: src/main/java/com/github/marciel404/bot/utils/loader.java
package com.github.marciel404.bot.utils;

import net.dv8tion.jda.api.JDA;
import net.dv8tion.jda.api.JDABuilder;
import net.dv8tion.jda.api.entities.Activity;
import net.dv8tion.jda.api.requests.GatewayIntent;
import net.dv8tion.jda.api.utils.ChunkingFilter;
import net.dv8tion.jda.api.utils.MemberCachePolicy;
import net.dv8tion.jda.api.utils.cache.CacheFlag;
import static com.github.marciel404.bot.utils.Configs.token;

public interface loader {
    JDA client = JDABuilder.createDefault(token)
            .addEventListeners(
                    new com.github.marciel404.bot.events.Ready(),
                    new com.github.marciel404.bot.events.Message(),
                    new com.github.marciel404.bot.events.Member(),
                    new com.github.marciel404.bot.commands.Actions(),
                    new com.github.marciel404.bot.eventbuttons.ActionsButtons())
            .enableIntents(
                    GatewayIntent.GUILD_MEMBERS,
                    GatewayIntent.DIRECT_MESSAGES,
                    GatewayIntent.MESSAGE_CONTENT
            )
            .disableCache(CacheFlag.ACTIVITY)
            .setMemberCachePolicy(MemberCachePolicy.VOICE.or(MemberCachePolicy.OWNER))
            .setChunkingFilter(ChunkingFilter.NONE)
            .disableIntents(
                    GatewayIntent.GUILD_PRESENCES,
                    GatewayIntent.GUILD_MESSAGE_TYPING
            )
            .setLargeThreshold(50)
            .setActivity(Activity.listening("Minhas engrenagens"))
            .build();
}

File: src/main/java/com/github/marciel404/bot/utils/loader.kt
package com.github.marciel404.bot.utils

import net.dv8tion.jda.api.JDABuilder
import net.dv8tion.jda.api.entities.Activity
import net.dv8tion.jda.api.requests.GatewayIntent
import net.dv8tion.jda.api.utils.ChunkingFilter
import net.dv8tion.jda.api.utils.MemberCachePolicy
import net.dv8tion.jda.api.utils.cache.CacheFlag


object Client{

    val client = JDABuilder.createDefault(Configs.token)
        .addEventListeners(
            com.github.marciel404.bot.events.Ready(),
            com.github.marciel404.bot.events.Message(),
            com.github.marciel404.bot.events.Member(),
            com.github.marciel404.bot.commands.Actions(),
            com.github.marciel404.bot.eventbuttons.ActionsButtons()
        )
        .disableCache(CacheFlag.ACTIVITY)
        .enableIntents(
            GatewayIntent.GUILD_MEMBERS,
            GatewayIntent.MESSAGE_CONTENT,
            GatewayIntent.GUILD_MESSAGES

        )
        .disableIntents(
            GatewayIntent.DIRECT_MESSAGES
        )
        .setMemberCachePolicy(MemberCachePolicy.VOICE.or(MemberCachePolicy.OWNER))
        .setChunkingFilter(ChunkingFilter.NONE)
        .setLargeThreshold(50)
        .setActivity(Activity.listening("Minhas engrenagens"))
        .build()
}